internal/app/networkd/pkg/networkd/misc.go

```diff
@@ -49,6 +52,14 @@ func writeResolvConf(resolvers []string) (err error) {
 		}
 	}
 
+	if domain, err := talosnet.DomainName(); err == nil {
+		if domain != "" {
+			if _, err = resolvconf.WriteString(fmt.Sprintf("domain %s\n", domain)); err != nil {
+				return errors.Wrap(err, "failed to add domain line to resolvconf")
```

Comment:
Can we make this `fmt.Errorf("failed to add domain line to resolvconf: %w", err)`